src/open_producten/utils/views.py

```diff
@@ -42,3 +46,43 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context.update({"component": self.component})
         return context
+
+
+class TranslatableViewSetMixin:
+    def get_requested_language(self):
+        accept_language = self.request.headers.get("Accept-Language")
+        if accept_language and "," in accept_language:
+            raise ParseError(
+                _("Only a single language in Accept-Language is supported.")
+            )
+
+        return accept_language or "nl"
+
+    def get_queryset(self):
+        language = self.get_requested_language()
```

Comment:
```suggestion
        from django.utils.translation import  get_language
        language = get_language()
```
Could replace with this if you use the `LocaleMiddleware`

Although it the queryset might just work without setting the language. It  does this in `django-model-translaiton` but I am not familiar with `django-parler` so IDK